crates/pop-fork/src/local.rs

```diff
@@ -181,15 +211,14 @@ impl LocalStorageLayer {
 			return Ok(cached.map(Arc::new));
 		}
 
-		// Case 3: Block before or at fork point - query remote directly
-		// First, get the block hash from the blocks table
-		let block = self.parent.cache().get_block_by_number(block_number).await?;
+		// Case 3: Block before or at fork point - fetch and cache block if needed
```

Comment:
We should only hit this if the user were to query for historical data prior to the fork height. 

If query is at fork point point the state should already exist. Am I right ?

Reply:
This does a better job at explaining what I had in mind: https://github.com/r0gue-io/pop-cli/pull/855#discussion_r2642939717